src/Tizen.NUI/src/public/WindowSystem/QuickPanelService.cs

```diff
@@ -0,0 +1,174 @@
+﻿/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace Tizen.NUI.WindowSystem.Shell
+{
+    /// <summary>
+    /// Class for the Tizen quickpanel service.
+    /// </summary>
+    /// This class is need to be hidden as inhouse API.
+    /// <since_tizen> 8 </since_tizen>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class QuickPanelService : IDisposable
+    {
+        private TizenShell _tzsh;
+        private IntPtr _tzshQpService;
+        private int _tzshWin;
+        private bool disposed = false;
+
+        /// <summary>
+        /// QuickPanel Type.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum QuickPanelType
+        {
+            /// <summary>
+            /// Unknown
+            /// </summary>
+            Unknown = 0x0,
+            /// <summary>
+            /// System default
+            /// </summary>
+            SystemDefault = 0x1,
+            /// <summary>
+            /// Context menu
+            /// </summary>
+            ContextMenu = 0x2,
+        }
+
+        /// <summary>
+        /// Creates a new Quickpanel Service handle.
+        /// </summary>
+        /// <param name="tzShell">The TzShell instance.</param>
+        /// <param name="win">The window to provide service of the quickpanel.</param>
+        /// <param name="type">The type of quickpanel service.</param>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when a argument is null.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public QuickPanelService(TizenShell tzShell, Window win, QuickPanelType type)
+        {
+            if (tzShell == null)
+            {
+                throw new ArgumentNullException((string)"tzShell");
```

Comment:
you don't need to cast again with the same type.

Reply:
I'm not be familiar with C# ..
If I remove that unmeaningful cast, visual studio note that is internal warning
"Represents text as a sequence of UTF-16 code units"
and visual studio recommend using nameof(tzShell) instead of ("tzShell")
So I added this cast (string) to remove warning.

Which way is better way..?

Reply:
I think you can ignore that warning when using const strings in English. 
Can you share the exactly warning code? e.g) CS####

Reply:
below warning is this warning.
Is it okay to remove this cast?
https://docs.microsoft.com/ko-kr/visualstudio/code-quality/ca1507?view=vs-2019

Reply:
or.. below comment you added use nameof()
As same, in this case using nameof() is right?